controllers/withwatcher/watcher_controller_helper_test.go

```diff
@@ -80,54 +85,58 @@ func createWatcherCR(managerInstanceName string, statusOnly bool) *v1alpha1.Watc
 				fmt.Sprintf("%s-watchable", managerInstanceName): "true",
 			},
 			Field: field,
+			Gateway: &v1alpha1.GatewayConfig{
+				LabelSelector: v1alpha1.DefaultIstioGatewaySelector(),
+			},
 		},
 	}
 }
 
-func listTestWatcherCrs(kcpClient client.Client) []*v1alpha1.Watcher {
+func listTestWatcherCrs() []*v1alpha1.Watcher {
+	watcherList := &v1alpha1.WatcherList{}
 	watchers := make([]*v1alpha1.Watcher, 0)
-	for _, component := range centralComponents {
-		watcherCR := &v1alpha1.Watcher{}
-		err := kcpClient.Get(suiteCtx, client.ObjectKey{
-			Name:      fmt.Sprintf("%s-sample", component),
-			Namespace: metav1.NamespaceDefault,
-		}, watcherCR)
-		if err != nil {
-			continue
-		}
-
-		watchers = append(watchers, watcherCR)
+	if err := controlPlaneClient.List(suiteCtx, watcherList); err != nil {
+		return watchers
```

Comment:
this should imo not be ignored to avoid flaky tests

Reply:
agree